spec/BowlingSpec.js

```diff
@@ -0,0 +1,186 @@
+describe('Bowling', function(){
+
+  beforeEach(function() {
+    bowling = new Bowling();
+  });
+
+  describe('Basic Scoring', function() {
+
+    it('knows which frame it is on', function() {
+      expect(bowling.bowlingFrame).toEqual(1);
+    });
+
+    it('can read back the scores for all frames', function() {
+      array = bowling.allFramesScore();
+      expect(array).toEqual([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+    });
+
+    it('can make one bowl and have the correct array returned', function() {
+      bowling.bowl(3);
+      array = bowling.allFramesScore();
+      expect(array).toEqual([3, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+    });
+
+    it('can give a final score', function(){
+      expect(bowling.finalScore()).toEqual(0);
+    });
+
+  });
+
+  describe('Frames', function() {
+    it('there should be ten frames made', function() {
+      var framesNumber = Object.keys(bowling.frames).length
+      var failures = 0
+      for (i = 1; i <= framesNumber; i++) {
+        if(bowling.frames[i] instanceof Frame){
+          continue;
+        } else {
+          failures++;
+        };
+      };
+      expect(failures).toEqual(0);
+      expect(framesNumber).toEqual(10);
+    });
```

Comment:
This is overkill - you're testing JS internals! Just check for the framesNumber, then you also only have 1 expectation